geforcenow.com: [macOS & iPadOS] The wrong values for are reported for the `MouseEvent.button`& `MouseEvent.which` property for extra mouse buttons.

lilyspiniolas opened the pull request:
#### 5aee38fcd3aefb448e6f3961878ebcbc7393c420
<pre>
geforcenow.com: [macOS &amp; iPadOS] The wrong values for are reported for the `MouseEvent.button`&amp; `MouseEvent.which` property for extra mouse buttons.
<a href="https://bugs.webkit.org/show_bug.cgi?id=280755">https://bugs.webkit.org/show_bug.cgi?id=280755</a>
<a href="https://rdar.apple.com/137592588">rdar://137592588</a>

Reviewed by Abrar Rahman Protyasha.

`MouseEvent.button` and `MouseEvent.which` now return the correct values for
mouse events occuring as a result of a fourth or fifth mouse button being
pressed. Thanks to Abrar Protyasha for starting this patch, and for the work
in 289800@main which is re-landed here as the foundation for these changes.

`MouseEvent.button` and `MouseEvent.which` now return the correct value when
the wheel button is clicked on iOS (macOS already supported this button).

Tested by `fast/events/mouse-event-buttons.html` on non-iOS platforms, and
tested on iOS by new API test `iOSMouseSupport.NonPrimaryMouseButtonsForEvents`.
Added an extra, unrelated test
`iOSMouseSupport.LeftClickFiresPointerAndMouseEvents` to increase test
comprehensiveness for iOS mouse support.

Tests: fast/events/mouse-event-buttons.html
       Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm

* LayoutTests/fast/events/fire-mousedown-while-pressing-mouse-button.html:
* LayoutTests/fast/events/mouse-click-events-expected.txt:
* LayoutTests/fast/events/mouse-click-events.html:
* LayoutTests/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/fast/events/mouse-event-buttons.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_click_on_chorded_mouse_button.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&amp;button=1&amp;buttons=4-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&amp;button=1&amp;buttons=4-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEventTypes.h:
(WebCore::buttonFromShort):
(WebCore::nsEventButtonNumberFromWebCoreMouseButton):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::currentMouseButton):
(WebCore::buttonFromButtonNumber):
(WebCore::mouseButtonForEvent):
* Source/WebKit/Shared/API/c/WKEvent.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
* Source/WebKit/Shared/WebMouseEvent.h:
* Source/WebKit/Shared/ios/AdditionalButtonMasksIOS.h: Added.
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebKit::WebIOSEventFactory::toUIEventButtonMask):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::setState):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/Automation.json:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::WebAutomationSession):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::mouseButtonToGdkButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionLibWPE.cpp:
(WebKit::mouseButtonToWPEButton):
(WebKit::stateModifierForWPEButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp:
(WebKit::libWPEMouseButtonToWPEButton):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):
* Tools/DumpRenderTree/mac/EventSendingController.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(eventTypeForMouseButtonAndAction):
(swizzledEventPressedMouseButtons):
(swizzledEventButtonNumber):
(-[EventSendingController mouseDown:withModifiers:]):
(-[EventSendingController mouseUp:withModifiers:]):
(-[EventSendingController contextClick]):
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(-[TestWKWebView pointerInfoAtLocation:]):
((iOSMouseSupport, TrackButtonMaskFromTouchStart)):
((iOSMouseSupport, MouseTimestampTimebase)):
((iOSMouseSupport, EndedTouchesTriggerClick)):
((iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
((iOSMouseSupport, MouseDidMoveOverElement)):
((iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)):
((iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)):
((iOSMouseSupport, ShowingContextMenuSelectsEditableText)):
((iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)):
((iOSMouseSupport, ContextClickAtEndOfSelection)):
((iOSMouseSupport, WebsiteMouseEventPolicies)):
((iOSMouseSupport, MouseInitiallyDisconnected)):
((iOSMouseSupport, MouseInitiallyConnected)):
((iOSMouseSupport, MouseLaterDisconnected)):
((iOSMouseSupport, MouseLaterConnected)):
((iOSMouseSupport, MouseAlwaysConnected)):
((iOSMouseSupport, BasicPointerInteractionRegions)):
((iOSMouseSupport, LeftClickFiresPointerAndMouseEvents)):
((iOSMouseSupport, NonPrimaryMouseButtonsForEvents)):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
* Tools/TestWebKitAPI/mac/PlatformWebViewMac.mm:
(TestWebKitAPI::eventTypeForButton):
(TestWebKitAPI::PlatformWebView::simulateButtonClick):
* Tools/WebKitTestRunner/EventSenderProxy.h:
(WTR::EventSenderProxy::lastButtonDown const):
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const): Deleted.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/EventSenderProxyCocoa.mm: Copied from Source/WebCore/dom/MouseEventTypes.h.
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const):
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::eventSenderButtonToGDKButton):
(WTR::stateModifierForGdkButton):
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:
(WTR::eventTypeForMouseButtonAndAction):
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::swizzledEventButtonNumber):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::sendMouseDownToStartPressureEvents):
(): Deleted.

Canonical link: <a href="https://commits.webkit.org/304537@main">https://commits.webkit.org/304537@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7e4afbbd0de29e378841995687fc0b28d9277bfb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/135913 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/8284 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/47199 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/143624 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/87472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ba558a23-2735-4e0d-a842-6efacd5d1c93) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/137782 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/8943 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/8128 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/103869 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/87472 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/138859 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/6471 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/121807 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/84746 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b067f383-0123-402d-a162-9fd9f41efa0d) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/6175 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/3819 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/4228 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/115440 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/39995 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/146373 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/7967 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/40561 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/112221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/7989 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/6690 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/112610 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28571 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/164/builds/6083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/118105 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/61865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/8014 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/36186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/7743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/71566 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/7963 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/7825 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
```diff
@@ -77,9 +78,11 @@
+    case WebKit::WebMouseEventButton::Back:
+        return WebKit::UIEventButtonMaskQuaternary;
+    case WebKit::WebMouseEventButton::Forward:
+        return WebKit::UIEventButtonMaskQuinary;
```
this is in `namespace WebKit`, so you can skip the prefixes for the button masks.

aprotyas replied:
I suppose you could keep the prefixes on `WebMouseEventButton` for visual parity

aprotyas commented on Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
```diff
@@ -3179,6 +3187,12 @@ void webkitWebViewBaseSynthesizeMouseEvent(WebKitWebViewBase* webViewBase, Mouse
+    case 8:
+        webEventButton = WebMouseEventButton::Back;
+        break;
+    case 9:
```
do you know where this comes from? just for my own reference, I could not find it in any documentation

lilyspiniolas replied:
I saw it in discussions online, will try to either verify locally or find some source code which proves it

aprotyas commented on Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
```diff
@@ -260,23 +261,46 @@ inline static String pointerType(UITouchType type)
+            : currentButtonMask;
+
+        if (isControlClick || (buttonMaskToCheck & UIEventButtonMaskSecondary))
+            buttonsBitmask |= 2;
```
nit: reorder this for the ascending bitmask consistency? 😄

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -796,6 +797,124 @@ static void simulateEditContextMenuAppearance(TestWKWebView *webView, CGPoint lo
+    testHarness.mouseDown(UIEventButtonMaskPrimary);
+    testHarness.mouseUp();
+
+    TestWebKitAPI::Util::runFor(0.1_s);
```
I suppose `[webView waitForPendingMouseEvents];` did not work here?

lilyspiniolas replied:
That's correct, I had to call `[webView waitForPendingMouseEvents]` in addition to `TestWebKitAPI::Util::runFor(0.1_s);` for the test to complete

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -796,6 +797,124 @@ static void simulateEditContextMenuAppearance(TestWKWebView *webView, CGPoint lo
+    // FIXME: It is not ideal that we have to wait for a static amount of time
+    // between sets of events if we want to test multiple pointerdowns.
+
+    auto webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```
```suggestion
    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
```

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -796,6 +797,124 @@ static void simulateEditContextMenuAppearance(TestWKWebView *webView, CGPoint lo
+    // between sets of events if we want to test multiple pointerdowns.
+
+    auto webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
```
```suggestion
    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
```

aprotyas commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
```diff
@@ -796,6 +797,124 @@ static void simulateEditContextMenuAppearance(TestWKWebView *webView, CGPoint lo
+    // between sets of events if we want to test multiple pointerdowns.
+
+    auto webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
```
```suggestion
    RetainPtr webViewConfiguration = adoptNS([[WKWebViewConfiguration alloc] init]);
    RetainPtr webView = adoptNS([[TestWKWebView alloc] initWithFrame:NSMakeRect(0, 0, 800, 600) configuration:webViewConfiguration.get()]);
```

aprotyas reviewed (approved):
LGTM, great patch, just one question about the static wait for the mouse harness.

lilyspiniolas committed:
geforcenow.com: [macOS & iPadOS] The wrong values for are reported for the `MouseEvent.button`& `MouseEvent.which` property for extra mouse buttons.
https://bugs.webkit.org/show_bug.cgi?id=280755
rdar://137592588

Reviewed by Abrar Rahman Protyasha.

`MouseEvent.button` and `MouseEvent.which` now return the correct values for
mouse events occuring as a result of a fourth or fifth mouse button being
pressed. Thanks to Abrar Protyasha for starting this patch, and for the work
in 289800@main which is re-landed here as the foundation for these changes.

`MouseEvent.button` and `MouseEvent.which` now return the correct value when
the wheel button is clicked on iOS (macOS already supported this button).

Tested by `fast/events/mouse-event-buttons.html` on non-iOS platforms, and
tested on iOS by new API test `iOSMouseSupport.NonPrimaryMouseButtonsForEvents`.
Added an extra, unrelated test
`iOSMouseSupport.LeftClickFiresPointerAndMouseEvents` to increase test
comprehensiveness for iOS mouse support.

Tests: fast/events/mouse-event-buttons.html
       Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm

* LayoutTests/fast/events/fire-mousedown-while-pressing-mouse-button.html:
* LayoutTests/fast/events/mouse-click-events-expected.txt:
* LayoutTests/fast/events/mouse-click-events.html:
* LayoutTests/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/fast/events/mouse-event-buttons.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_mouse-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_attributes_pen-right-nonstandard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_click_on_chorded_mouse_button.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&button=1&buttons=4-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/fast/events/mouse-event-buttons-expected.txt: Added.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/uievents/mouse/synthetic-mouse-enter-leave-over-out-button-state-after-target-removed.tentative_buttonType=MIDDLE&button=1&buttons=4-expected.txt:
* LayoutTests/platform/win/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* Source/WebCore/dom/MouseEvent.cpp:
(WebCore::MouseEvent::button const):
* Source/WebCore/dom/MouseEventTypes.h:
(WebCore::buttonFromShort):
(WebCore::nsEventButtonNumberFromWebCoreMouseButton):
* Source/WebCore/platform/mac/PlatformEventFactoryMac.mm:
(WebCore::currentMouseButton):
(WebCore::buttonFromButtonNumber):
(WebCore::mouseButtonForEvent):
* Source/WebKit/Shared/API/c/WKEvent.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
* Source/WebKit/Shared/WebEvent.serialization.in:
* Source/WebKit/Shared/WebEventConversion.cpp:
(WebKit::platform):
(WebKit::kit):
* Source/WebKit/Shared/WebMouseEvent.h:
* Source/WebKit/Shared/ios/AdditionalButtonMasksIOS.h: Added.
* Source/WebKit/Shared/ios/WebIOSEventFactory.mm:
(WebKit::WebIOSEventFactory::toUIEventButtonMask):
* Source/WebKit/Shared/mac/WebEventFactory.mm:
(WebKit::WebEventFactory::toNSButtonNumber):
* Source/WebKit/UIProcess/API/glib/WebKitPrivate.cpp:
(toWebKitMouseButton):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(MotionEvent::setState):
(webkitWebViewBaseSynthesizeMouseEvent):
* Source/WebKit/UIProcess/Automation/Automation.json:
* Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp:
(WebKit::WebAutomationSession::WebAutomationSession):
* Source/WebKit/UIProcess/Automation/gtk/WebAutomationSessionGtk.cpp:
(WebKit::mouseButtonToGdkButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionLibWPE.cpp:
(WebKit::mouseButtonToWPEButton):
(WebKit::stateModifierForWPEButton):
* Source/WebKit/UIProcess/Automation/libwpe/WebAutomationSessionWPE.cpp:
(WebKit::libWPEMouseButtonToWPEButton):
* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::sendSynthesizedEventsToPage):
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):
* Source/WebKit/UIProcess/ios/WKMouseInteraction.mm:
(-[WKMouseInteraction createMouseEventWithType:wasCancelled:]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):
* Tools/DumpRenderTree/mac/EventSendingController.h:
* Tools/DumpRenderTree/mac/EventSendingController.mm:
(eventTypeForMouseButtonAndAction):
(swizzledEventPressedMouseButtons):
(swizzledEventButtonNumber):
(-[EventSendingController mouseDown:withModifiers:]):
(-[EventSendingController mouseUp:withModifiers:]):
(-[EventSendingController contextClick]):
* Tools/TestWebKitAPI/Configurations/Base.xcconfig:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/iOSMouseSupport.mm:
(-[TestWKWebView pointerInfoAtLocation:]):
((iOSMouseSupport, TrackButtonMaskFromTouchStart)):
((iOSMouseSupport, MouseTimestampTimebase)):
((iOSMouseSupport, EndedTouchesTriggerClick)):
((iOSMouseSupport, CancelledTouchesDoNotTriggerClick)):
((iOSMouseSupport, MouseDidMoveOverElement)):
((iOSMouseSupport, SelectionUpdatesBeforeContextMenuAppears)):
((iOSMouseSupport, DisablingTextIteractionPreventsSelectionWhenShowingContextMenu)):
((iOSMouseSupport, ShowingContextMenuSelectsEditableText)):
((iOSMouseSupport, ShowingContextMenuSelectsNonEditableText)):
((iOSMouseSupport, ContextClickAtEndOfSelection)):
((iOSMouseSupport, WebsiteMouseEventPolicies)):
((iOSMouseSupport, MouseInitiallyDisconnected)):
((iOSMouseSupport, MouseInitiallyConnected)):
((iOSMouseSupport, MouseLaterDisconnected)):
((iOSMouseSupport, MouseLaterConnected)):
((iOSMouseSupport, MouseAlwaysConnected)):
((iOSMouseSupport, BasicPointerInteractionRegions)):
((iOSMouseSupport, LeftClickFiresPointerAndMouseEvents)):
((iOSMouseSupport, NonPrimaryMouseButtonsForEvents)):
* Tools/TestWebKitAPI/Tests/mac/MenuTypesForMouseEvents.mm:
(TestWebKitAPI::buildAndPerformTest):
* Tools/TestWebKitAPI/mac/PlatformWebViewMac.mm:
(TestWebKitAPI::eventTypeForButton):
(TestWebKitAPI::PlatformWebView::simulateButtonClick):
* Tools/WebKitTestRunner/EventSenderProxy.h:
(WTR::EventSenderProxy::lastButtonDown const):
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const): Deleted.
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/EventSenderProxyCocoa.mm: Copied from Source/WebCore/dom/MouseEventTypes.h.
(WTR::EventSenderProxy::mouseButtonsCurrentlyDown const):
* Tools/WebKitTestRunner/gtk/EventSenderProxyGtk.cpp:
(WTR::eventSenderButtonToGDKButton):
(WTR::stateModifierForGdkButton):
* Tools/WebKitTestRunner/mac/EventSenderProxy.mm:
(WTR::eventTypeForMouseButtonAndAction):
(WTR::EventSenderProxy::EventSenderProxy):
(WTR::swizzledEventButtonNumber):
(WTR::EventSenderProxy::mouseDown):
(WTR::EventSenderProxy::mouseUp):
(WTR::EventSenderProxy::sendMouseDownToStartPressureEvents):
(): Deleted.

Canonical link: https://commits.webkit.org/304537@main
